### ISSUE TITLE
Hazard WRITE_AFTER_WRITE using Secondary Command Buffers

### ISSUE THREAD

**Issue by wusikijeronii** [KhronosGroup/Vulkan-ValidationLayers#7743]
**Environment:**
 - OS: Windows 11 22H2
 - GPU and driver version: NVIDIA GeForce RTX 3050 Laptop. Driver version: 551.86.0.0
 - SDK or header version if building from repo: 1.3.280.0
 - Options enabled (synchronization, best practices, etc.): synchronization

**Describe the Issue**
The source issue was described on [Discourse](https://community.khronos.org/t/hazard-write-after-write/110622). 
I faced the problem when I used multisubpass and secondary command buffers. I got the Hazard WAW error. Without secondary buffers, I got no errors.
You can familiarize yourself with the example based on VulkanTutorial [here](https://github.com/wusikijeronii/VulkanTutorial-cleaned/blob/main/code/27_depth_buffering.cpp).
I received an error about sync image layout, but it is somewhat nonsensical that Store Op could hazard anything (look at the error below).

**Additional context**

<details>
<summary>code or terminal output</summary>

```sh
validation layer: Validation Error: [ SYNC-HAZARD-WRITE-AFTER-WRITE ] Object 0: handle = 0xead9370000000008, type = VK_OBJECT_TYPE_RENDER_PASS; | MessageID = 0x5c0ec5d6 | vkCmdEndRenderPass():  Hazard WRITE_AFTER_WRITE in subpass 1 for attachment 1 color aspect during store with storeOp VK_ATTACHMENT_STORE_OP_DONT_CARE. Access info (usage: SYNC_COLOR_ATTACHMENT_OUTPUT_COLOR_ATTACHMENT_WRITE, prior_usage: SYNC_IMAGE_LAYOUT_TRANSITION, write_barriers: SYNC_FRAGMENT_SHADER_INPUT_ATTACHMENT_READ, command: vkCmdNextSubpass, seq_no: 3, subcmd: 1, renderpass: VkRenderPass 0xead9370000000008[], reset_no: 2)
```

</details>

**Comment by artem-lunarg** [comment]
I can build the original sample and reproduce the error. That should be a validation bug. For some reason the layout transition does not synchronizes correctly with the accesses from the secondary command buffer. The net effect that the layout transition detects only accesses after the secondary command buffer (DONT_CARE storeOp writes). Then the system thinks there is a missing synchronization with DONT_CARE access.

Repro cases:
https://github.com/artem-lunarg/Vulkan-ValidationLayers/commit/6f139ec46de5e0dc8b6ecf99bf677f4d6584d99a

The first test reproduces the issue in a single subpass render pass by using LoadOp NONE.
The second test reproduces the same issue in a two subpass scenario, it's more similar to the reported sample.

**Comment by artem-lunarg** [comment]
TLDR; It can be limitation of current syncval design.

I will think if we can avoid this error message without implementing quite non-trivial solution, but not sure if it can be done reliably.

The problem is that we can't track attachment accesses when recording secondary command buffers. That's because we don't know `renderArea`. In contrast to to core checks, the syncval is not designed to postpone such validation to submit time. The goal of curernt syncval's submit time validation is to check accesses on the boundaries of submitted command buffers. But we did not consider that a separate command can't be validated by processing its commands one at a time. 

Specifically this code:
```
if (sync_index == SYNC_FRAGMENT_SHADER_INPUT_ATTACHMENT_READ) {
    const VkExtent3D extent = CastTo3D(cb_state_->render_area.extent);
    const VkOffset3D offset = CastTo3D(cb_state_->render_area.offset);
    current_context_->UpdateAccessState(*img_view_state, sync_index, SyncOrdering::kRaster, offset, extent,
                                        tag_ex);
}
```
`cb_state_->render_area` is a garbage value for secondary command buffers that record render pass commands. So far it looks like in order to support such validation we need some serious extension of syncval functionality, like an option to postpone validation during record time and instead serialize all commands and then replay them during ExecuteCommands. Currently we do replay for syncval submit time validation, but that replay is about replaying sync operations (barriers), we don't have support for replaying arbitrary commands.

**Comment by artem-lunarg** [comment]
Updated tests to work with the latest test framework: https://github.com/artem-lunarg/Vulkan-ValidationLayers/commit/af63e7045d7c3739e6058abeb1ae7b6d1c475d44
